std-bits/table/src/main/java/org/enso/table/excel/ExcelUtils.java

```diff
@@ -0,0 +1,71 @@
+package org.enso.table.excel;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.Temporal;
+
+public class ExcelUtils {
+  private static final LocalDate EPOCH_1900 = LocalDate.of(1899, 12, 30);
+  private static final long MILLIS_PER_DAY = 24 * 60 * 60 * 1000L;
+
+  /** Converts an Excel date-time value to a {@link Temporal}. */
+  public static Temporal fromExcelDateTime(double value) {
+    // Excel treats 1900-02-29 as a valid date, which it is not a valid date.
+    if (value >= 60 && value < 61) {
+      return null;
+    }
+
+    // For days before 1900-01-01, Stored as milliseconds before 1900-01-01.
+    long days = (long) value;
+
+    // Extract the milliseconds part of the value.
+    long millis = (long) ((value - days) * MILLIS_PER_DAY + (value < 0 ? -0.5 : 0.5));
+    if (millis < 0) {
+      millis += MILLIS_PER_DAY;
+    }
+    if (millis != 0 && value < 0) {
+      days--;
+    }
+
+    // Excel stores times as 0 to 1.
+    if (days == 0) {
+      return LocalTime.ofNanoOfDay(millis * 1000000);
+    }
+
+    // Deal with mess of Jan and Feb 1900.
+    // In order to avoid issues with 1900-01-00, 1899-12-31 is -1.
+    int shift = days > 0 && days < 60 ? 1 : (days < 0 ? 2 : 0);
```

Comment:
minor style preference: IMHO it could be more readable if we do it as
```suggestion
    int shift = 0;
    if (days > 0 && days < 60) {
        // some comment about first 60 days
        shift = 1;
    } else if (days < 0) {
        // some comment
        shift = 2;
    }
```

What do you think? It makes it possible to follow and document the edge cases a bit more carefully.

But feel free to keep as is, as it's OK too.